stubs/openpyxl/openpyxl/workbook/workbook.pyi

```diff
@@ -20,54 +31,58 @@ class Workbook:
-    def epoch(self, value) -> None: ...
+    def epoch(self, value: datetime.datetime) -> None: ...
     @property
-    def read_only(self): ...
+    def read_only(self) -> bool: ...
     @property
-    def data_only(self): ...
+    def data_only(self) -> bool: ...
     @property
-    def write_only(self): ...
+    def write_only(self) -> bool: ...
     @property
-    def excel_base_date(self): ...
+    def excel_base_date(self) -> datetime.datetime: ...
     @property
-    def active(self): ...
+    def active(self) -> Worksheet: ...
     @active.setter
-    def active(self, value) -> None: ...
-    def create_sheet(self, title: Incomplete | None = ..., index: Incomplete | None = ...): ...
-    def move_sheet(self, sheet, offset: int = ...) -> None: ...
-    def remove(self, worksheet) -> None: ...
+    def active(self, value: Worksheet) -> None: ...
+    def create_sheet(self, title: str | _Decodable | None = ..., index: int | None = ...) -> WriteOnlyWorksheet | Worksheet: ...
+    def move_sheet(self, sheet: str | Worksheet, offset: int = ...) -> None: ...
+    def remove(self, worksheet: Worksheet | WriteOnlyWorksheet | Chartsheet) -> None: ...
     def remove_sheet(self, worksheet) -> None: ...
-    def create_chartsheet(self, title: Incomplete | None = ..., index: Incomplete | None = ...): ...
+    def create_chartsheet(self, title: str | _Decodable | None = ..., index: int | None = ...): ...
     def get_sheet_by_name(self, name): ...
-    def __contains__(self, key): ...
-    def index(self, worksheet): ...
+    def __contains__(self, key: str): ...
+    def index(self, worksheet: Worksheet | ReadOnlyWorksheet | WriteOnlyWorksheet) -> int: ...
     def get_index(self, worksheet): ...
-    def __getitem__(self, key): ...
-    def __delitem__(self, key) -> None: ...
+    def __getitem__(self, key: str): ...
+    def __delitem__(self, key: str) -> None: ...
     def __iter__(self): ...
     def get_sheet_names(self): ...
     @property
-    def worksheets(self): ...
+    def worksheets(self) -> list[Worksheet | ReadOnlyWorksheet | WriteOnlyWorksheet]: ...
     @property
-    def chartsheets(self): ...
+    def chartsheets(self) -> list[Chartsheet]: ...
     @property
-    def sheetnames(self): ...
+    def sheetnames(self) -> list[str]: ...
     def create_named_range(
-        self, name, worksheet: Incomplete | None = ..., value: Incomplete | None = ..., scope: Incomplete | None = ...
+        self,
+        name: str | None,
+        worksheet: _WorkbookChild | ReadOnlyWorksheet | None = ...,
+        value: str | None = ...,
+        scope: int | None = ...,
     ) -> None: ...
-    def add_named_style(self, style) -> None: ...
+    def add_named_style(self, style: NamedStyle) -> None: ...
     @property
-    def named_styles(self): ...
-    def get_named_ranges(self): ...
-    def add_named_range(self, named_range) -> None: ...
+    def named_styles(self) -> list[str]: ...
+    def get_named_ranges(self) -> list[DefinedName]: ...
```

Comment:
Conflict: See https://github.com/python/typeshed/pull/9764#discussion_r1113412107

Reply:
Your rationale in https://github.com/python/typeshed/pull/9764#discussion_r1113412107 is convincing; let's go with a `list`